File: Python/1863.py
from typing import List

class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:

        def dfs(i, ans):
            if i == len(nums): return ans
            return dfs(i+1, ans^nums[i]) + dfs(i+1, ans)
               
        return dfs(0, 0)

# Topic: Dfs, Bit Manipulation
# Time Complexity: O(2^n)
# Space Complexity: O(n)

File: C++/1863.cpp
#include <vector>

class Solution {
public:
    int dfs(int i, int ans, std::vector<int>& nums){
        if (i == nums.size()) return ans;
        return dfs(i+1, ans^nums[i], nums) + dfs(i+1, ans, nums);
    }
    int subsetXORSum(std::vector<int>& nums) {
        return dfs(0, 0, nums);        
    }
};

// Topic: Bit Manipulation, DFS
// Time Complexity: O(2^n)
// Space Complexity: O(n)